tests/test_foodfidelity_groups.py

```diff
@@ -36,20 +42,20 @@ def test_ingredients(self):
         self.assertEqual(
             [
                 "1 cup water",
-                "2/3 cup orange juice",
-                "1 1/3 cups pearl couscous",
+                "2/3 cupÂ orange juice",
```

Comment:
Let's try to figure out whatever problematic encoding has occurred here - from a few quick searches, I _think_ it's something related to non-breaking spaces in whatever original content source these ingredient lists were sourced from.  Either way, we don't want the `Â` characters (nor do we want to turn them into spaces, because that'd result in two-space separators between words).

Reply:
It's something to do with a [UTF-8 encoded non-breaking space](https://en.wikipedia.org/wiki/Non-breaking_space), I think..

```pyi
>>> '\xc2\xa0'
'Â\xa0'
```

Reply:
I'm able to resolve it with a function in the scraper class that does some replacements

```python
    def clean_ingredient(ingredient):
        unescaped_ingredient = html.unescape(ingredient)
        cleaned_ingredient = re.sub(r'[\xa0\u200b\xc2]', ' ', unescaped_ingredient)
        cleaned_ingredient = re.sub(r'\s+', ' ', cleaned_ingredient).strip()
        return cleaned_ingredient

    def ingredients(self):
        ingredients_cleaned = self.schema.ingredients()
        return [self.clean_ingredient(ingredient) for ingredient in ingredients_cleaned]
```

I think this is a pretty unique case so i think it can be handled right in the class.

If this looks good i'll implement & fix the tests.

Reply:
Thanks - it's debatable I guess, particulraly because the added `.replace` call could be invoked a _lot_, but I've added handling for this #897 as you noticed.

I don't think we're at the point where we need to fine-tune the string operations performed by individual scrapers, so I think that performing the replacement in the shared utility method _probably_ makes sense for now.